### DISCUSSION TITLE
Proper endpoint to exempt to successfully post dataset via API??

### DISCUSSION THREAD

**Discussion by tduval-unifylogic** [apache/superset#19356] [General]
regarding issue [#17206](https://github.com/apache/superset/issues/17206)

I am, like [LeiZhanglmo](https://github.com/LeiZhangImo) attempting to create a dataset via the api to a table that already exists.
I have followed the guidance for setting 
I am using apache-superset-1.4.1rc1
 
I have set *WTF_CSRF_ENABLED = False* in *config.py* and have added (what I believe may be incorrect) the endpoint to the *WTF_CSRF_EXEMPT_LIST* so I won't face the notorius: "CSRF Token is missing" error:
![image](https://user-images.githubusercontent.com/62360524/159812386-72d7c4de-67a1-4b9c-a350-1ed9c9543913.png)

Alas, to no avail! 

I am unsuccessful via the Swagger API page as well as attempting to post in my notebook [provided by Srini Kadamati](https://www.youtube.com/watch?v=fyfAXAbxhg0)

<img width="1040" alt="image" src="https://user-images.githubusercontent.com/62360524/159813183-cc05b50e-6aff-40d5-9d51-061329f2bc44.png">

Can someone please respond with what the proper endpoint address is to put in the list so that we can successfully add datasets via the api???  I am sure this is obvious to some, but it doesn't appear to be jumping out to me. 

Thanks a lot!

**Comment by AAfghahi** [comment]
Hello, 

I think that this might help:

https://github.com/apache/superset/issues/16003

**Reply by AAfghahi** [reply]
I also converted this to a discussion since I do not think that it is a bug.

**Reply by tduval-unifylogic** [reply]
Please restore to bug due to the following:

1. I am posting to the dataset/post API exactly as depicted in the discussion you referred me to. I am using the very same CSRF token I used successfully on a previous api call and receive CSRF Token is missing. To me, that indicates the API does not work as intended. That to me is a bug. The discussion is helpful, but I have already covered this ground and am at an impasse due to it. 

This is EXACTLY what I am doing:
   ![image](https://user-images.githubusercontent.com/62360524/160034301-79a79686-b8ce-4afc-85a4-4d0b72fdc88f.jpeg)

I even intercepted the POST with Postman from the SS form and still receive the same error.
	
If it makes a difference, I am using the Docker version of Superset. Thought I would mention if it does. 

3. Also, It could be me, but Superset (1.4.1rc) does not appear to honor setting WTF_CSRF_ENABLED flag to False (I still get the CSRF token is missing after setting and restarting SS with config). Note: I would rather not disable this setting as we are looking to use SS in a product, therefore, back to item 1

 4. Also, my question also in relation to which endpoint path I should put in the list of exempt endpoints. If that could be provided, that would help me with a workaround until this issue is resolved.

Just in case, tomorrow I will attempt the same call, with the same parameters on two other instances of SS on two different servers in an attempt to reproduce the issue...I am hoping it is just a PEBCAK (Problem Exists Between Chair and Keyboard) Issue.

Thanks in advance.

**Reply by tduval-unifylogic** [reply]
@[AAfghahi](https://github.com/AAfghahi)

I have tested again today the issue and found it is repeatable. The post below (that fails) was recorded by postman interceptor from the SS api Create Dataset UI (and was successful btw) 

Please reply with any: 

1. Possible work arounds (e.g., exclusions in config.py for endpoint)
2. Reproduction of this issue
3. Suggested remedies

I am needing this endpoint to function as designed since creating a dataset via REST API is crucial to an upcoming release for my team.

Thanks!!


Unsuccessful Post in Notebook:
<img width="728" alt="image" src="https://user-images.githubusercontent.com/62360524/160210563-10bffd3f-232a-4a92-9597-62ab03da2442.png">

Unsuccessful Post in Postman:
<img width="554" alt="image" src="https://user-images.githubusercontent.com/62360524/160210369-e3b8727a-8dbc-40c4-9c95-ecb16c64069d.png">

**Comment by ahsanshah** [comment]
I am facing the same issue. in my case I am running dbt superset lineage but get the exact same issue.  Is there a bug with Superset which simply prevents the WTF_CSRF_ENABLED = False from being invoked?  I confirmed my ENV is set properly.  Note that I am also using Docker compose, etc.
